src/bin/sol-fbp-runner/main.c

```diff
@@ -149,6 +172,180 @@ parse_args(int argc, char *argv[])
+                                SOL_WRN("Couldn't get entry name");
+                                goto error;
+                            }
+                        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "offset")) {
+                            if (sol_json_token_get_uint32(&value, &offset) < 0) {
+                                SOL_WRN("Couldn't get entry offset");
+                                goto error;
+                            }
+                        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "size")) {
+                            if (sol_json_token_get_uint32(&value, &size) < 0) {
+                                SOL_WRN("Couldn't get entry size");
+                                goto error;
+                            }
+                        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "bit_offset")) {
+                            if (sol_json_token_get_uint32(&value, &bit_offset) < 0) {
+                                SOL_WRN("Couldn't get entry size");
+                                goto error;
+                            }
+                            if (bit_offset > 7) {
+                                SOL_WRN("Entry bit offset cannot be greater than 7");
+                                goto error;
+                            }
+                        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "is_bool")) {
+                            is_bool = sol_json_token_get_type(&token) == SOL_JSON_TYPE_TRUE;
+                        }
+                    }
+
+                    ptr_table_entry = sol_vector_append(&entries_vector);
+                    SOL_NULL_CHECK_GOTO(ptr_table_entry, error);
+
+                    memmap_entry = calloc(sizeof(struct sol_memmap_entry), 1);
+                    SOL_NULL_CHECK_GOTO(memmap_entry, error);
+
+                    memmap_entry->offset = offset;
+                    memmap_entry->is_bool = is_bool;
+                    if (is_bool) {
+                        memmap_entry->bit_offset = bit_offset;
+                        memmap_entry->size = sizeof(uint8_t);
+                    } else {
+                        if (!size) {
+                            SOL_WRN("Invalid size for entry");
+                            free(memmap_entry);
+                            goto error;
+                        }
+                        memmap_entry->size = size;
+                    }
+
+                    ptr_table_entry->key = strdup(name);
+                    ptr_table_entry->len = strlen(name);
+                    ptr_table_entry->val = memmap_entry;
+                }
+            }
+        }
+
+        /* Add ptr_table guard element */
+        ptr_table_entry = sol_vector_append(&entries_vector);
+        SOL_NULL_CHECK_GOTO(ptr_table_entry, error);
+
+        entries_vector_size = sizeof(struct sol_str_table_ptr) * entries_vector.len;
+        map = calloc(sizeof(struct sol_memmap_map) + entries_vector_size, 1);
```

Comment:
Suggestion: what about making `struct sol_memmap_map` abstract and using something like 
`struct sol_memmap_map *sol_memmap_map_new(const char path, struct sol_str_table_ptr table[])` to create a new instance, then the only place where you would check for `USE_MEMMAP` would be there.

Would this work?